src/NServiceBus.AcceptanceTesting/ScenarioWithContext.cs

```diff
@@ -54,6 +55,17 @@ public Task<TContext> Run(TimeSpan? testExecutionTimeout = null)
             return WithEndpoint<T>(b => { });
         }
 
+        public IScenarioWithEndpointBehavior<TContext> WithCustomComponent(
+            string name,
+            Func<TContext, CancellationToken, Task> onStart, 
+            Func<TContext, Task> onStop, 
+            Func<TContext, CancellationToken, Task> onStarted = null) 
```

Comment:
also, how about this method just accepting a `IComponentBehavior` parameter? Then you could provide your own implementation in your repositories and we don't need to include them in the att framework?